src/pages/MainPage.js

```diff
@@ -1,10 +1,103 @@
+import styled from '@emotion/styled'
 import { Header, Banner } from '@components'
+import { useState, useEffect } from 'react'
+import { getChannelList, getPostListInChannel, getChannelInfo, getSearchedBookList } from '@apis'
+import { BookListSlider, Input, Button, Select } from '@components'
+
+const ALL_CATEGORY = 'all'
+const DEFAULT_CATEGORY = '소설'
+
+const SearchBar = styled.div`
+  display: flex;
+  justify-content: center;
+`
 
 const MainPage = () => {
+  const [postList, setPostList] = useState([])
+  const [categoryName, setCategoryName] = useState(ALL_CATEGORY)
+
+  const [searchKeyword, setSearchKeyword] = useState('')
+
+  const fetchAllPost = async () => {
+    const channelList = await getChannelList()
+    const totalPostList = await Promise.all(
+      channelList.map(async (channel) => await getPostListInChannel(channel._id))
+    ).then((res) => res.flat())
```

Comment:
```suggestion
      channelList.map(async (channel) =>{
         const res = await getPostListInChannel(channel._id))
         return res.flat()
      })
    )
```
위의 코드는 정확한 코드는 아니지만 위의 다은님께서 제게 말씀해주신 것 처럼 안티패턴이라 위의 예시처럼 수정하면 좋을 것 같습니다!!